app/jobs/reservation_job.rb

```diff
@@ -0,0 +1,44 @@
+class ReservationJob < ActiveJob::Base
```

Comment:
I just wanted to say that I still think this is awesome, great use of inheritance IMO.